File: docs/static/examples/src/noise/c++/noise.cpp
#include "p8g.hpp"
#include "SimplexNoise.h"
#include <cmath>

using namespace p8g;

float r = 0.f;

void p8g::draw() {
  background(255);
  for (int y = 0; y < height; y++) {
    stroke(255 * SimplexNoise::noise(y));
    for (int x = 0; x < width; x++) {
      point(99 * SimplexNoise::noise((x - y) / 99.f + r) * sinf((x + y) / 99.f) + y, x);
    }
  }
  r -= M_PI / 99.f;
}

void p8g::keyPressed() {}
void p8g::keyReleased() {}
void p8g::mouseMoved() {}
void p8g::mousePressed() {}
void p8g::mouseReleased() {}
void p8g::mouseWheel(float delta) {}

int main() {
  run(320, 320, "Noise");
}

File: docs/static/examples/src/noise/js/noise.js
import p8g, {
  background,
  createCanvas,
  height,
  point,
  stroke,
  width,
} from 'p8g.js';
import { createNoise2D } from 'simplex-noise';

const noise2D = createNoise2D();
let r = 0;

p8g.draw = () => {
  background(255);
  for (let y = 0; y < height; y++) {
    stroke(255 * noise2D(y, 0));
    for (let x = 0; x < width; x++) {
      point(99 * noise2D((x - y) / 99 + r, 0) * Math.sin((x + y) / 99) + y, x);
    }
  }
  r -= Math.PI / 99;
};

createCanvas(320, 320);
